pkg/flowaggregator/exporter/ipfix.go

```diff
@@ -119,24 +125,37 @@ func (e *IPFIXExporter) AddRecord(record ipfixentities.Record, isRecordIPv6 bool
 	return nil
 }
 
-func (e *IPFIXExporter) updateExternalFlowCollectorAddr(address, protocol string) {
-	if address == e.externalFlowCollectorAddr && protocol == e.externalFlowCollectorProto {
+func (e *IPFIXExporter) UpdateOptions(opt *options.Options) {
+	config := opt.Config.FlowCollector
+	if reflect.DeepEqual(config, e.config) && opt.Config.RecordContents.PodLabels == e.includePodLabels {
```

Comment:
It seems unsupported changes could return false here and leads to "New IPFIXExporter configuration" log even nothing changes, for example, by updating `enable`.  Could it be confusing?

According to the description "updates to recordContents.podLabels and recordFormat were ignored, without any kind of warning.", should it warn users that it's unsupported?

Reply:
If there is an update to `enable`, this code will never be executed: https://github.com/antrea-io/antrea/blob/264372dc33003a997e92acb8dbd5dded0c5278a8/pkg/flowaggregator/flowaggregator.go#L618-L634
All other fields in `opt.Config.FlowCollector` are now handled correctly with this PR, which is why I used `reflect.DeepEqual` on the struct. I took a similar approach for the "LogExporter": https://github.com/antrea-io/antrea/blob/264372dc33003a997e92acb8dbd5dded0c5278a8/pkg/flowaggregator/exporter/log.go#L150-L161

One could argue that this code is a bit clumsy. It may not have been a great idea to support live config updates for the Flow Aggregator in the first place...

> According to the description "updates to recordContents.podLabels and recordFormat were ignored, without any kind of warning.", should it warn users that it's unsupported?

With this PR, these updates should now be supported.

Reply:
Got it, thanks for the explanation.